### PR TITLE
add math support when using JavaFX

### PR DESCRIPTION
**Pull request by aplatanado**
This is my try to add math support to JavaFX preview, following #69 requirements.
So, MathJax installation is an offline installation and no CDN.

I have included a submodule to MathJax 2.7.2 because the installation instructions say most files in the repository are require.

The only issues is I don't know get the stem setting from JavaFxHtmlPanel, so MathJax is always included.

### CODE REVIEW

**Comment by ahus1** [comment]
Thanks for this PR. Using the git submodule can be a choice. AFAIK this automatically points to the master of MathJax and not a specific version. AFAIK git submodules can't point to a tag, and that's a pity. But let's don't worry about that now. 

If we would build the plugin it will include everything from the MathJax directory. That would add white some Megabytes to the plugin. 

I added a change to build.gradle to exclude files that might not be needed from looking at file names and .npmignore in the MathJax folder.

Maybe you can exclude some more files. I don't have any test cases, please post some here (this would be a great help). 

Thanks!

File: src/main/java/org/asciidoc/intellij/editor/javafx/JavaFxHtmlPanel.java [hunk 1]
```diff
@@ -61,6 +61,11 @@ protected String compute() {
         .append("<script src=\"").append(clazz.getResource("scrollToElement.js")).append("\"></script>\n")
         .append("<script src=\"").append(clazz.getResource("processLinks.js")).append("\"></script>\n")
         .append("<script src=\"").append(clazz.getResource("pickSourceLine.js")).append("\"></script>\n")
+        .append("<script src=\"").append(clazz.getResource("mathJaxConfig.js")).append("\"></script>\n")
```
Comment (ahus1) [inline src/main/java/org/asciidoc/intellij/editor/javafx/JavaFxHtmlPanel.java]: I didn't find mathJaxConfig.js here, only localMathJaxConfig.js. Could mathJaxConfig.js be removed, of the other file renamed?

Comment (aplatanado) [reply src/main/java/org/asciidoc/intellij/editor/javafx/JavaFxHtmlPanel.java]: Ups, L64 is a remain of my attempts to load the MathJax configuration from a file.
I thought I got it using the config URL option but I was wrong. After remove L64 (a line with no sense), MathJax doesn't work.

So, I have made a commit where the content of localMathJaxConfig.js is in-line.

**Comment by aplatanado** [comment]
This is MathJax sample-asciimath.html in asciidoc notation:

~~~~
:stem:

When stem:[a != 0], there are two solutions to stem[ax^2 + bx + c = 0] and they are

[stem]
++++
x = (-b +- sqrt(b^2-4ac))/(2a)
++++
~~~~

**Comment by aplatanado** [comment]
I found this tool to minify a MathJax installation:

https://github.com/zackw/minify-mathjax/blob/master/minify.cfg

I will try it.

**Comment by aplatanado** [comment]
I have removed the submodule and included a minified MathJax installation made with minify-mathjax. It occupy 8Mb.

**Comment by ahus1** [comment]
Thanks for the contribution! It will take some more days until I get access to a fast internet connection to try it out.

**Comment by ahus1** [comment]
@aplatanado - reviewed minify.cfg and made two changes: 
* Asciidoctor 1.5.x uses MathJax 2.6, therefore I downgraded. I want the plugin to be in sync (just in case)
* JavaFxHtmlPanel included TeX-MML-AM_HTMLorMML (like asciidoctor does) - therefore I changes minify.cfg to do the same. 

I found out about the second bullet point only after I did a clean in my workspace.

Please give this version a quick tryout (I don't have math in my docs). I will then squash-and-merge the change.

Thanks again for this PR!

**Comment by aplatanado** [comment]
Thank you @ahus1.
I have added extension/asciimath2jax.js because it is required by TeX-MML-AM_HTMLorMML.

**Comment by ahus1** [comment]
@pantelis - I suppose you commented on the Asciidoc plugin homepage and asked for Math support. Since 0.20.0 this is now supported.

**Pull request merged**